plutus-core/cost-model/budgeting-bench/Bench.hs

```diff
@@ -1,154 +1,254 @@
+
+benchTwoIntegers :: StdGen -> DefaultFun -> Benchmark
+benchTwoIntegers gen builtinName =
+    createTwoTermBuiltinBench builtinName inputs inputs
+    where
+      (numbers,_) = makeDefaultIntegerArgs gen
+      inputs  = fmap (\e -> (e, memoryUsage e)) numbers
+
+benchSameTwoIntegers :: StdGen -> DefaultFun -> Benchmark
+benchSameTwoIntegers gen builtinName = createTwoTermBuiltinBenchElementwise builtinName inputs inputs'
+    where
+      (numbers,_) = makeDefaultIntegerArgs gen
+      inputs  = fmap (\e -> (e, memoryUsage e)) numbers
+      inputs' = fmap (\e -> (e, memoryUsage e)) $ clone numbers
+          where clone = (fmap (\n -> (n-1) `div` 3) . fmap (\n -> 3*n+1))
+    -- Let's try to make sure that the numbers aren't physically identical;
+    -- hopefully this is sufficiently complicated that GHC won't be able to
+    -- spot that it's the identity function.
+
+
+---------------- Bytestring builtins ----------------
+integerPower :: Integer -> Integer -> Integer
+integerPower = (^) -- Just to avoid some type ascriptions later
+
+seedA :: HH.Seed
+seedA = HH.Seed 42 43
+
+seedB :: HH.Seed
+seedB = HH.Seed 44 45
+
+-- TODO: we're using Hedgehog for some things and System.Random for others.  We
+-- should rationalise this.  Pehaps Hedgehog is more future-proof since it can
+-- produce random instances of a wide variety of types.
+genSample :: HH.Seed -> HH.Gen a -> a
+genSample seed gen = Prelude.maybe (Prelude.error "Couldn't create a sample") HH.treeValue $ HH.evalGen (HH.Size 1) seed gen
+
+byteStringSizes :: [Integer]
+byteStringSizes = integerPower 2 <$> [1..20::Integer]
+
+makeSizedBytestring :: HH.Seed -> Int -> (BS.ByteString, ExMemory)
+makeSizedBytestring seed e = let x = genSample seed (HH.bytes (HH.Range.singleton e)) in (x, memoryUsage x)
+
+byteStringsToBench :: HH.Seed -> [(BS.ByteString, ExMemory)]
+byteStringsToBench seed = (makeSizedBytestring seed . fromInteger) <$> byteStringSizes
 
 benchHashOperations :: DefaultFun -> Benchmark
 benchHashOperations name =
     bgroup (show name) $
-        byteStringsToBench seedA <&> (\(x, xMem) ->
-            runTermBench (show xMem) $ erase $ mkIterApp () (builtin () name) [(mkConstant () x)]
-        )
+        byteStringsToBench seedA <&> (\(x, xmem) -> runTermBench (show xmem) $ mkApp1 name x)
+
+benchTwoByteStrings :: DefaultFun -> Benchmark
+benchTwoByteStrings name = createTwoTermBuiltinBench name (byteStringsToBench seedA) (byteStringsToBench seedB)
+
+-- Copy the bytestring here, because otherwise it'll be exactly the same, and the equality will short-circuit.
+benchSameTwoByteStrings :: DefaultFun -> Benchmark
+benchSameTwoByteStrings name = createTwoTermBuiltinBenchElementwise name (byteStringsToBench seedA)
+                               ((\(bs, e) -> (BS.copy bs, e)) <$> byteStringsToBench seedA)
+
+powersOfTwo :: [Integer]
+powersOfTwo = integerPower 2 <$> [1..16] -- <> ((\(a :: Integer) -> 10^a) <$> [3..8])
```

Comment:
Commented out code.

Reply:
I'll get rid of that.  I think that's left over from way back.